service/storage_service.cc

```diff
@@ -4716,6 +4716,7 @@ future<> storage_service::decommission() {
 
                 slogger.info("DECOMMISSIONING: starts");
                 ctl.req.leaving_nodes = std::list<gms::inet_address>{endpoint};
+                ctl.req.endpoint_to_host_id[endpoint] = ctl.host_id;
```

Comment:
Nit: this is a round-about way to do it in C++ (`operator[]` to default-construct the object, then `operator=` to assign a different value to it); why not use the direct way i.e. `emplace` or `insert` methods?